README.md

```diff
@@ -26,62 +26,48 @@ Please return to this page regularly to stay updated with the progress of develo
 
 ## For Developers
 
-It is recommended to work on N-Trans in a virtual environment.
+N-Trans is split into three phases.
 
-To set up a virtual environment:
-- MAC OS:
-  `python3 -m venv env`
-
-To activate a virtual environment:
-- MAC OS:
-  `source env/bin/activate`
+1) In `ntrans_dataprep.py`, sentences from the BNC are processed and, split into N-grams, and written
+to .csv files in chunks of 300K sentences.
 
+The script generates a total of 236.937.946 N-grams.
 
+2) In `ntrans_compile.py`, the frequency of each N-gram is counted with `collections.Counter`. The 50K
+most frequent N-grams are thereafter written to a new .csv file. One file is created for each N-gram
+(2-grams, 3-grams, etc.)
 
-### Libraries
+3) `ntrans.py` is the main program which the end user is exposed to. Here, the X most frequent N-grams
+will be machine translated to a chosen target language, and the source-target pair is written to a .csv file.
 
+### Development
 
-#### NLTK (Docs: https://www.nltk.org/)
+It is recommended to work on N-Trans in a virtual environment.
 
-To install NLTK:
+To set up a virtual environment:
 - MAC OS:
-  `python3 -m pip install nltk`
-
-NLTK is used to process the data and create N-grams from it.
-
-
-#### Re (Docs: https://docs.python.org/3/library/re.html)
-
-Re is used to remove punctuation & special characters from the downloaded data
-
-#### collections.Counter (Docs: https://docs.python.org/3/library/collections.html)
-
-collections.Counter is used to count the frequency of each N-gram to distinguish the most common ones.
-
-#### translatepy (Docs: https://github.com/Animenosekai/translate)
+  `python3 -m venv env`
 
-To install translatepy:
+To activate a virtual environment:
 - MAC OS:
-  `python3 -m pip install translatepy`
-
-Translatepy is used to, with the help of several machine translation APIs, translate the X most common N-grams.
-
-
-#### CSV (Docs: https://docs.python.org/3/library/csv.html)
+  `source env/bin/activate`
 
-CSV is used to create the N-Trans glossary file.
+### Install Dependencies
 
+To install the dependencies needed to develop & test N-Trans, use:
 
-### Tools
+- MAC OS:
+  `python3 -m pip install -r requirements.txt`
 
+### Formatting
 
-#### Black (Docs: https://pypi.org/project/black/)
 This project uses Black to format the code. Please use Black before creating a PR.
 
-To install Black:
-- MAC OS:
-  `python3 -m pip install black`
-
 To use Black:
 - MAC OS:
-  `black ntrans.py`
+  `black file.py`
+
+### Module Docs
+
+NLTK: https://www.nltk.org/
+Translatepy: https://github.com/Animenosekai/translate
```

Comment:
I'm not sure if this section is needed, since any developer is used to googling things like "nltk docs" or "translatepy docs".